components/camera/camera.go

```diff
@@ -75,8 +77,78 @@ type Properties struct {
 
 // NamedImage is a struct that associates the source from where the image came from to the Image.
 type NamedImage struct {
-	Image      image.Image
+	data       []byte
+	img        image.Image
 	SourceName string
+	mimeType   string
+}
+
+// NamedImageFromBytes constructs a NamedImage from a byte slice, source name, and mime type.
+func NamedImageFromBytes(data []byte, sourceName, mimeType string) (NamedImage, error) {
+	if data == nil {
+		return NamedImage{}, fmt.Errorf("must provide image bytes to construct a named image from bytes")
+	}
+	if mimeType == "" {
+		return NamedImage{}, fmt.Errorf("must provide a mime type to construct a named image")
+	}
+	return NamedImage{data: data, SourceName: sourceName, mimeType: mimeType}, nil
+}
+
+// NamedImageFromImage constructs a NamedImage from an image.Image, source name, and mime type.
+func NamedImageFromImage(img image.Image, sourceName, mimeType string) (NamedImage, error) {
+	if img == nil {
+		return NamedImage{}, fmt.Errorf("must provide image to construct a named image from image")
+	}
+	if mimeType == "" {
+		return NamedImage{}, fmt.Errorf("must provide a mime type to construct a named image")
+	}
+	return NamedImage{img: img, SourceName: sourceName, mimeType: mimeType}, nil
+}
+
+// Image returns the image.Image of the NamedImage.
+func (ni *NamedImage) Image(ctx context.Context) (image.Image, error) {
+	if ni.img == nil {
+		if ni.data == nil {
+			return nil, fmt.Errorf("no image or image bytes available")
+		}
+
+		reader := bytes.NewReader(ni.data)
+		_, header, err := image.DecodeConfig(reader)
+		if err != nil {
+			return nil, fmt.Errorf("could not decode image config: %w", err)
+		}
+
+		if header != "" && !strings.Contains(ni.mimeType, header) {
+			return nil, fmt.Errorf("mime type does not match the image bytes: expected %s, got %s", ni.mimeType, header)
```

Comment:
Why would we not want to update the mimeType to ground truth and return success in the case they do not match?

Sorry if this was already hashed out in the scope doc.

Reply:
[`ErrMIMETypeBytesMismatch](https://github.com/viamrobotics/rdk/pull/5183/files#diff-4eab9b93978c09650defc23c4274a4c77d87cfdea7775d86dcf8c838ec207a01R134)` diff looks good